core/clustersmngr/factory_caches.go

```diff
@@ -28,6 +30,18 @@ func (c *Clusters) Get() []Cluster {
 	return c.clusters
 }
 
+func (c *Clusters) Hash(clusters []Cluster) string {
```

Comment:
I wonder why accept a list of clusters instead of just accessing them directly and returning the hash?